test/types/schema.test.ts

```diff
@@ -863,6 +862,19 @@ function gh12242() {
   expectType<0 | 1>({} as Example['active']);
 }
 
+function testInferTimestamps() {
+  const schema = new Schema({
+    name: String
+  }, { timestamps: true });
+
+  type WithTimestamps = InferSchemaType<typeof schema>;
+  // For some reason, expectType<{ createdAt: Date, updatedAt: Date, name?: string }> throws
```

Comment:
that is because the type literally is a AND between multiple things and the internal representation (as you can see with a hover in vscode) and `expectType` is strict and only matches the actual internal representation, if you want the simpler notion use `expectAssignable`

Reply:
Thanks for clarifying, I spent an hour or two trying to figure out WTF was going on :face_with_spiral_eyes: